pkg/build/registry/buildconfig/webhook.go

```diff
@@ -72,9 +73,13 @@ func (w *WebHook) ServeHTTP(writer http.ResponseWriter, req *http.Request, ctx k
 		Revision:    revision,
 		Env:         envvars,
 	}
-	if _, err := w.instantiator.Instantiate(config.Namespace, request); err != nil {
+	newBuild, err := w.instantiator.Instantiate(config.Namespace, request)
+	if err != nil {
 		return errors.NewInternalError(fmt.Errorf("could not generate a build: %v", err))
 	}
+	// Send back the build name so that the client can alert the user.
+	newBuildJson, _ := json.Marshal(newBuild)
```

Comment:
> the caller is expecting nothing today. Returning a status object that includes the build (if nothing went wrong) sounds like the most flexible course of action.

I intended to agree with what you said before about the encoding error case: you shouldn't return unknown content.

Returning a build object (as long as its an external version) seems reasonable for success cases.  Returning a status object for encoding failure cases is ok.  Most of our API already behaves that way.